geonode/static/geonode/js/upload/upload.js

```diff
@@ -352,6 +352,10 @@ define(['underscore',
             buildFileInfo(_.groupBy(file_input.files, path.getName));
             displayFiles(file_queue);
         });
+        // Detect click on "Remove" link and update the file_queue
+        $(options.file_queue).on('click', 'a', function () {
+            displayFiles(file_queue);
+        });
```

Comment:
Refresh the display of all layers when a file is removed, this way we can handle when a layer has been deleted.

I'm open for suggestions on better ways to handle this case, but I'm not sure it's possible to achieve this without delegating responsibility to upload.js.

Perhaps there is a way to detect a deletion of an object in layers and respond to that?